src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -306,25 +338,50 @@ def disable_project_uptime_subscription(uptime_monitor: ProjectUptimeSubscriptio
     # All project subscriptions are disabled, we can disable the subscription
     # and remove the remote subscription.
     if not has_active_subscription:
-        uptime_subscription.update(status=UptimeSubscription.Status.DISABLED.value)
+        uptime_subscription.status = UptimeSubscription.Status.DISABLED.value
+        uptime_subscription.save()
         delete_remote_uptime_subscription.delay(uptime_subscription.id)
 
 
-def enable_project_uptime_subscription(uptime_monitor: ProjectUptimeSubscription):
+def enable_project_uptime_subscription(
+    uptime_monitor: ProjectUptimeSubscription, ensure_assignment: bool = False
+):
     """
-    Re-enables a project uptime subscription. If the uptime subscription was
+    Enable a project uptime subscription. If the uptime subscription was
     also disabled it will be re-activated and the remote subscription will be
     published.
+
+    This method will attempt seat assignment via the quotas system. If There
+    are no available seats the monitor will be disabled and a
+    `UptimeMonitorNoSeatAvailable` will be raised.
+
+    By default if the monitor is already marked as ACTIVE this function is a
+    no-op. Pass `ensure_assignment=True` to force seat assignment.
     """
-    if uptime_monitor.status != ObjectStatus.DISABLED:
+    if not ensure_assignment and uptime_monitor.status != ObjectStatus.DISABLED:
         return
 
-    uptime_monitor.update(status=ObjectStatus.ACTIVE)
+    seat_assignment = quotas.backend.check_assign_seat(DataCategory.UPTIME, uptime_monitor)
```

Comment:
So if we're forcing, and this seat is already assigned and there are no more free seats, will this have a valid seat assignment?

Reply:
hmm. If we're passing `ensure_assignment` it means we've creating a new monitor, so there will be no seat.

yeah not sure about this api either 🤔

Reply:
This is confusing since it's called by something named `get_or_create` but it's actually only ever using it to create afaict